test/Conversion/accelerate-matmul.mlir

```diff
@@ -0,0 +1,16 @@
+// RUN: triton-opt %s --convert-triton-to-tritongpu --tritongpu-accelerate-matmul=compute-capability=80 -tritongpu-remove-layout-conversions -tritongpu-optimize-dot-operands -tritongpu-remove-layout-conversions -tritongpu-reduce-data-duplication -allocate-shared-memory  -convert-triton-gpu-to-llvm
```

Comment:
Do you need all of these passes?  If the crash is in accelerate-matmul, presumably you only need that one pass, you can run triton-to-tritongpu beforehand and check in that result.

Reply:
I can create a testcase with ttgir as input.
These passes are needed because the wrong op is created in accelerate-matmul, but it crashes in convet-tritonggpu-to-llvm. Without these passes, compiler crashes in other ways......
For example, https://github.com/openai/triton/blob/f08bdc1a8ac846ba876aa0e1882c2459ac698b6a/lib/Dialect/TritonGPU/IR/Dialect.cpp#L1911 here seems to only consider rank-2 shapes, not rank-3 (compare with here https://github.com/openai/triton/blob/f08bdc1a8ac846ba876aa0e1882c2459ac698b6a/lib/Dialect/TritonGPU/IR/Dialect.cpp#L1891)

Reply:
Under TritonGPU, there's also a `accelerate-matmul.mlir` file

Reply:
> Under TritonGPU, there's also a `accelerate-matmul.mlir` file

This bug only crashes when convert-tritongpu-to-llvm. The accelerate-matmul.mlir file under TritonGPU only run accelerate-matmul, it needs more passes before convert-tritongpu-to-llvm, otherwise it crashes......Therefore I put this testcase somewhere else.

Reply:
the lit test should run just the pass you are changing and you need to add CHECK to verify that the code you changed do what you expected or in the case of a negative test that the transformation did not happen. This allows testing isolated part of the code.
You should be able to get the input of the pass based on your example, then you just need to add the right checks. And as Keren mentioned please move it in accelerate-matmul.mlir

Reply:
the testcase has been revised.